transport/src/main/java/com/strategyobject/substrateclient/transport/ws/WsProvider.java

```diff
@@ -299,6 +296,24 @@ public Runnable on(ProviderInterfaceEmitted type, EventListener sub) {
         return whenUnsubscribed;
     }
 
+    private void scheduleCleanupIfNoResponseWithinTimeout(int id) {
+        timedOutHandlerCleaner.schedule(() -> {
+                    if (!this.handlers.containsKey(id)) {
```

Comment:
Might be an issue here when timeout and response happen at the same time. So cleanup and response handler might race to complete the target future. It might be better to do
```
val handler = this.handlers.remove(id);
if (handler == null) {
    return;
}

handler.getCallback()
    .completeExceptionally(new TimeoutException(
        String.format("The node didn't respond for %s milliseconds.",
            responseTimeoutInMs)));
```

Reply:
Yep, thanks!